### PR TITLE
Update from Rust 1.43 to 1.46 and improve packaging in minor ways

### PR DESCRIPTION
**Pull request by thombles**
In particular, toolchains are now installed against their specific version number so they can coexist. This is particularly useful in CI environments where different branches may be on different versions.

**The good news:** The rust hacks were upstreamed so we no longer need to patch rustc, nor do we need any `-Z` flags. Since the `-Z` flag is no longer needed, we are now building the stable channel.

**The bad news:** There is no "good" LLVM commit to target due to a bug. Here we are targeting the 5.3-RELEASE tag, which should match Xcode 12 pretty closely, and using a patch to allow the rust compilation to proceed. See the comments in the patch for a full explanation.

I have confirmed that Rust code built with this toolchain can be included in bitcode apps, deployed to physical hardware, and rebuilt from bitcode.

### CODE REVIEW

File: config.sh [hunk 1]
```diff
@@ -5,15 +5,14 @@
 # Apple Swift version 5.1.3 (swiftlang-1100.0.282.1 clang-1100.0.33.15)
```
Comment (hamchapman) [inline config.sh:5]: The info here (and just above and below) looks out of date. Maybe we don't need the specifics at all anymore?

File: dist.sh [hunk 2]
```diff
@@ -2,25 +2,35 @@
 set -euxo
 source config.sh
 
+# The built toolchain that we are going to package
 WORKING_DIR="$(pwd)/build"
-DEST="$(pwd)/dist/rust-ios-arm64-${RUST_TOOLCHAIN}"
-TOOLCHAIN_DEST="${DEST}/toolchain-${RUST_TOOLCHAIN}"
 
-rm -rf "$TOOLCHAIN_DEST"
+# The directory which will added to the final zip file
```
Comment (hamchapman) [inline dist.sh:8]:
```suggestion
# The directory which will be added to the final zip file
```

**Review by hamchapman — APPROVED** [review]
Couple of documentation/comment nits but LGTM!

Nice job getting to something that works 🙌

**Review by konstantinbe — APPROVED** [review]
Nice, LGTM. 👍

**Review by connorpower — APPROVED** [review]
LGTM 👍

**Pull request merged**